src/arithimpl/gmpimpl.rs

```diff
@@ -5,21 +5,31 @@ extern crate gmp;
 use super::traits::*;
 use self::gmp::mpz::Mpz;
 use self::gmp::rand::RandState;
+use rand::{OsRng, Rng};
 
 impl Samplable for Mpz {
+
     fn sample_below(upper: &Self) -> Self {
-        let mut r = RandState::new();
-        r.urandom(upper)
-    }
+        let mut state = RandState::new();
```

Comment:
the Python phe package avoids GMP randomness: https://github.com/NICTA/python-paillier/blob/master/phe/paillier.py#L146 -- not sure what that means though